src/BeaverBuilderFieldsFilterHelper.php

```diff
@@ -0,0 +1,249 @@
+            foreach ($remove as $regex) {
+                if (0 !== preg_match("~$regex~", $key)) {
+                    continue 2;
+                }
+            }
+            $result[$key] = $value;
+        }
+
+        return $result;
+    }
+
+    public function applyTranslatedValues(SubmissionEntity $submission, array $originalValues, array $translatedValues, $_ = true): array
+    {
+        $result = array_merge(
+            $this->flattenArray($this->prepareSourceData($originalValues)),
+            $this->flattenArray($this->prepareSourceData($translatedValues)),
+        );
+
+        try {
+            $removeFields = Bootstrap::getContainer()->get('content-serialization.helper')->getRemoveFields();
+        } catch (\Exception $e) {
+            $removeFields = [];
+        }
+        if (!array_key_exists('entity', $removeFields)) {
+            $removeFields['entity'] = [];
+        }
+        $removeFields['entity'] = array_merge($removeFields['entity'], ['ID', 'post_status', 'guid', 'comment_count']);
+        foreach ($removeFields as $prefix => $fields) {
+            foreach ($fields as $field) {
+                unset ($result["$prefix/$field"]);
+            }
+        }
+
+        return $this->inflateArray(get_post_meta($submission->getSourceId(), '_fl_builder_data')[0] ?? [], $result);
+    }
+
+    private function buildData(\stdClass $original, array $array, string $prefix, string $path = ''): \stdClass
+    {
+        $arrayOriginal = (array)$original;
+        foreach ($array as $key => $value) {
+            $currentType = gettype($value);
+            $newPath = ltrim($path . self::ARRAY_DIVIDER . $key, self::ARRAY_DIVIDER);
+            if (array_key_exists($prefix . $newPath, $arrayOriginal)) {
+                $originalType = gettype($arrayOriginal["$prefix$newPath"]);
+                if ($currentType !== $originalType) {
+                    if (is_array($value) && $originalType === 'object') {
+                        $array[$key] = $this->buildData($original, $value, $prefix, $newPath);
+                    } elseif (is_scalar($value)) {
+                        settype($array[$key], $originalType);
+                    }
+                }
+            }
+        }
+
+        return (object)$array;
+    }
+
+    private function inflateArray(array $data, array $translated): array
+    {
+        $result = $this->structurizeArray($translated, self::ARRAY_DIVIDER);
+        foreach ($result['meta'][self::DATA_FIELD_NAME] as $key => $value) {
+            $result['meta'][self::DATA_FIELD_NAME][$key] = $this->buildData(
+                $data[$key],
+                $value,
+                ''
+            );
+        }
+        $result['meta']['_fl_builder_data_settings'] = $this->toStdClass($result['meta']['_fl_builder_data_settings']);
+        return $result;
+    }
+
+    public function flattenArray(array $array, string $base = '', string $divider = self::ARRAY_DIVIDER): array
```

Comment:
Maybe it should be private?

Reply:
It can't be made private, unfortunately, as \Smartling\Helpers\FieldsFilterHelper::flattenArray is public